lib/barbeque/message_handler/job_execution.rb

```diff
@@ -6,19 +6,21 @@ module Barbeque
   module MessageHandler
     class JobExecution
       # @param [Barbeque::Message::JobExecution] message
-      # @param [Barbeque::JobQueue] job_queue
-      def initialize(message:, job_queue:)
+      # @param [Barbeque::MessageQueue] message_queue
+      def initialize(message:, message_queue:)
         @message = message
-        @job_queue = job_queue
+        @message_queue = message_queue
       end
 
       def run
         begin
-          job_execution = Barbeque::JobExecution.create(message_id: @message.id, job_definition: job_definition, job_queue: @job_queue)
+          job_execution = Barbeque::JobExecution.create(message_id: @message.id, job_definition: job_definition, job_queue: @message_queue.job_queue)
         rescue ActiveRecord::RecordNotUnique => e
+          @message_queue.delete_message(@message)
```

Comment:
In the following flow,

* worker1: `Barbeque::JobExecution.create` => success
* worker2: `Barbeque::JobExecution.create` for the same message => fail
* worker1: immediate shutdown before `Barbeque::ExecutionLog.save_message`
* worker2: deletes message in this line

In this case, message is deleted before saving message to S3. Is this intentional? I don't think we need to delete message in this line.